esp/services/ws_dali/ws_daliservice.cpp

```diff
@@ -244,3 +243,222 @@ bool CWSDaliEx::onCount(IEspContext& context, IEspCountRequest& req, IEspCountRe
     }
     return true;
 }
+
+IUserDescriptor* CWSDaliEx::createUserDesc(IEspContext& context)
+{
+    StringBuffer username;
+    context.getUserID(username);
+    if (username.isEmpty())
+        return nullptr;
+
+    Owned<IUserDescriptor> userdesc = createUserDescriptor();
+    userdesc->set(username.str(), context.queryPassword(), context.querySignature());
+    return userdesc.getClear();
+}
+
+bool CWSDaliEx::onDFSLS(IEspContext& context, IEspDFSLSRequest& req, IEspResultResponse& resp)
+{
+    try
+    {
+        checkAccess(context);
+
+        StringBuffer options;
+        if (req.getRecursively())
+            options.append("r");
+        if (!req.getPathAndNameOnly())
+            options.append("l");
+        if (req.getIncludeSubFileInfo())
+            options.append("s");
+
+        StringBuffer result;
+        dfsLs(req.getName(), options, result);
+        resp.setResult(result);
+    }
+    catch(IException* e)
+    {
+        FORWARDEXCEPTION(context, e,  ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+bool CWSDaliEx::onGetDFSCSV(IEspContext& context, IEspGetDFSCSVRequest& req, IEspResultResponse& resp)
+{
+    try
+    {
+        checkAccess(context);
+
+        Owned<IUserDescriptor> userDesc = createUserDesc(context);
+
+        const char* dali = req.getDali();
+
+        StringBuffer result;
+        dfscsv(!isEmptyString(dali) ? dali : "*", userDesc, result);
```

Comment:
The "*" in daadmin.cpp looks wrong/odd.
The usage of dfscsv is:
```
  printf("  dfscsv <logicalnamemask>       -- get csv info. for files matching mask\n");
```
i.e. the 1 param should be used as a filter to files return, not as dali param.

I think would be ok to fix this at this time, i.e. here and in daadmin.cpp, rather than propagate the incorrect usage

Reply:
I will fix.